mlflow/transformers.py

```diff
@@ -0,0 +1,990 @@
+
+    if not isinstance(model, (TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel)):
+        raise MlflowException(
+            "The supplied model type is unsupported. The model must be one of: "
+            "PreTrainedModel, TFPreTrainedModel, or FlaxPreTrainedModel",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    base_reqs = ["transformers"]
+    try:
+        base_reqs.extend(_model_packages(model))
+        return [_get_pinned_requirement(module) for module in base_reqs]
+    except Exception as e:
+        dependencies = [
+            _get_pinned_requirement(module)
+            for module in ["transformers", "torch", "torchvision", "tensorflow"]
+        ]
+        _logger.warning(
+            "Could not infer model execution engine type due to huggingface_hub not "
+            "being installed or unable to connect in online mode. Adding full "
+            f"dependency chain: {dependencies}. \nFailure cause: {str(e)}"
+        )
+        return dependencies
+
+
+def _validate_transformers_model_dict(transformers_model):
+    """
+    Validator for a submitted save dictionary for the transformers model. If any additional keys
+    are provided, raise to indicate which invalid keys were submitted.
+    """
+    if isinstance(transformers_model, dict):
+        invalid_keys = [key for key in transformers_model.keys() if key not in _SUPPORTED_SAVE_KEYS]
+        if invalid_keys:
+            raise MlflowException(
+                "Invalid dictionary submitted for 'transformers_model'. The "
+                f"key(s) {invalid_keys} are not permitted. Must be one of: "
+                f"{_SUPPORTED_SAVE_KEYS}",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        if _MODEL_KEY not in transformers_model:
+            raise MlflowException(
+                f"The 'transformers_model' dictionary must have an entry for {_MODEL_KEY}",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        model = transformers_model[_MODEL_KEY]
+    else:
+        model = transformers_model.model
+    if not hasattr(model, "name_or_path"):
+        raise MlflowException(
+            f"The submitted model type {type(model).__name__} does not inherit "
+            "from a transformers pre-trained model. It is missing the attribute "
+            "'name_or_path'. Please verify that the model is a supported "
+            "transformers model.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+
+@experimental
+def get_default_conda_env(model):
+    """
+    :return: The default Conda environment for MLflow Models produced with the ``transformers``
+             flavor, based on the model instance framework type of the model to be logged.
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements(model))
+
+
+@experimental
+@docstring_version_compatibility_warning(integration_name=FLAVOR_NAME)
```

Comment:
Can we also apply this decorator to `load_model()`?

Reply:
Added! 
![Screen Shot 2023-03-29 at 7 05 57 PM](https://user-images.githubusercontent.com/39283302/228687662-62fe164f-142e-427a-94d2-24087ee002ad.png)